ee/hogai/graph/deep_research/report/nodes.py

```diff
@@ -89,8 +90,27 @@ async def arun(self, state: DeepResearchState, config: RunnableConfig) -> Partia
             context=context,
         )
 
+        notebook_title = self.notebook.title if self.notebook else "Research Report"
+        current_notebook_info = DeepResearchNotebookInfo(
+            stage=DeepResearchNodeName.REPORT.value,
+            notebook_id=notebook_update_message.notebook_id,
+            title=notebook_title,
+        )
+
+        # Collect all stage notebooks from the entire research process
+        # and add to report update message
+        all_stage_notebooks = [*state.stage_notebooks, current_notebook_info]
+
+        notebook_update_message.stage_notebooks = [
```

Comment:
`self._astream_notebook` returns a `NotebookUpdateMessage` with `notebook_id` (see the other comment about why that field is redundant now), which goes counter to what we're trying to do here, we should probably edit that function to just return a notebook, and then we construct the message outside of it.